syntax = "proto3";

package user;

import "google/protobuf/empty.proto";

service UserService {
  rpc CreateUser(CreateUserRequest) returns (UserInfoResponse);
  rpc GetUserById(UserIdRequest) returns (UserInfoResponse);
  rpc GetUserByMobile(UserMobileRequest) returns (UserInfoResponse);
  rpc UpdateUserAvatar(UserUpdateAvatarRequest) returns (google.protobuf.Empty);
  rpc UpdatePassword(UserUpdatePasswordRequest) returns (google.protobuf.Empty);
  rpc UpdateUsername(UserUpdateUsernameRequest) returns (google.protobuf.Empty);
  rpc GetUserList(PageRequest) returns (UserListResponse);
  rpc VerifyUser(VerifyUserRequest) returns (UserInfoResponse);
  rpc GetOrCreateUserByMobile(UserMobileRequest) returns (UserInfoResponse);
}

message CreateUserRequest {
  string mobile = 1;
}

message UserIdRequest {
  int64 id = 1;
}

message UserMobileRequest {
  string mobile = 1;
}

message UserUpdateAvatarRequest {
  int64 id = 1;
  string avatar = 2;
}

message UserUpdatePasswordRequest {
  int64 id = 1;
  string password = 2;
}

message UserUpdateUsernameRequest {
  int64 id = 1;
  string username = 2;
}

message PageRequest {
  int32 page = 1;
  int32 size = 2;
}

message VerifyUserRequest {
  string mobile = 1;
  string password = 2;
}

message UserInfo {
  int64 id = 1;
  string username = 2;
  string mobile = 3;
  string avatar = 4;
  bool is_activate = 5;
  bool is_staff = 6;
  string last_login = 7;
}

message UserInfoResponse {
  UserInfo user_info = 1;
}

message UserListResponse {
  repeated UserInfo user_list = 1;
}
